qiskit/circuit/random/utils.py

```diff
@@ -11,15 +11,415 @@
+
+                if is_cond:
+                    qc.measure(qc.qubits, cr)
+                    # The condition values are required to be bigints, not Numpy's fixed-width type.
+                    operation = operation.c_if(cr, int(condition_values[c_ptr]))
+                    c_ptr += 1
+
+                if control_qubit in qubit_idx_not_used and target_qubit in qubit_idx_not_used:
+                    qc._append(
+                        CircuitInstruction(
+                            operation=operation,
+                            qubits=[qubits[control_qubit], qubits[target_qubit]],
+                        )
+                    )
+                    qubit_idx_used.update(set(edge))
+                    qubit_idx_not_used = qubit_idx_not_used - qubit_idx_used
+                    edges_used[(control_qubit, target_qubit)] += 1
+
+            if insert_1q_oper:
+                if not len(qubit_idx_not_used) == 0:
+                    # Some extra 1Q Gate in to fill empty qubits for this while iteration.
+
+                    extra_1q_gates = rng.choice(
+                        gates_1q, size=len(qubit_idx_not_used), replace=True
+                    )
+                    cumsum_params = np.cumsum(extra_1q_gates["num_params"], dtype=np.int64)
+                    parameters_1q = rng.uniform(0, 2 * np.pi, size=cumsum_params[-1])
+
+                    for current_gate, num_gate_params, qubit_idx in zip(
+                        extra_1q_gates["class"], extra_1q_gates["num_params"], qubit_idx_not_used
+                    ):
+
+                        param = parameters_1q[:num_gate_params]
+                        parameters_1q = parameters_1q[num_gate_params:]
+                        qc._append(
+                            CircuitInstruction(
+                                operation=current_gate(*param),
+                                qubits=[qubits[qubit_idx]],
+                            )
+                        )
+
+        else:
+            for current_gate, num_gate_params, edge in zip(
+                gate_choices["class"],
+                gate_choices["num_params"],
+                edge_choices,
+            ):
+                params = parameters[:num_gate_params]
+                parameters = parameters[num_gate_params:]
+                operation = current_gate(*params)
+                control_qubit, target_qubit = tuple(edge)
+
+                if control_qubit in qubit_idx_not_used and target_qubit in qubit_idx_not_used:
+                    qc._append(
+                        CircuitInstruction(
+                            operation=operation,
+                            qubits=[qubits[control_qubit], qubits[target_qubit]],
+                        )
+                    )
+                    qubit_idx_used.update(set(edge))
+                    qubit_idx_not_used = qubit_idx_not_used - qubit_idx_used
+                    edges_used[(control_qubit, target_qubit)] += 1
```

Comment:
I wanted to acheive a do-while loop behavior.

Anyways, changed the code to you liking :)

Reply:
This would led to the calculation of the boolean list of conditionals, even if conditionals is not required :)
This would add a tiny bit of overhead which could have been avoided if this block doesn't run at all for conditional set to False.
That is why I had kept the two separate blocks :)

Anyways, change the code to your liking :)